csv/src/fs2/data/csv/CellDecoder.scala

```diff
@@ -53,25 +95,94 @@ object CellDecoder {
+    enum.values.find(_.toString === s).toRight(new DecoderError(s"unable to decode '$s' as a ${enum.toString()} value"))
+
+  // Primitives
+  implicit val unitDecoder: CellDecoder[Unit] = s =>
+    if (s.isEmpty)
+      Right(())
+    else
+      Left(new DecoderError(s"'$s' can't be decoded as unit as it is not empty"))
+  implicit val booleanDecoder: CellDecoder[Boolean] = s =>
+    Either.catchNonFatal(s.toBoolean).leftMap(new DecoderError(s"unable to decode '$s' as a byte", _))
+  implicit val byteDecoder: CellDecoder[Byte] = s =>
+    Either.catchNonFatal(s.toByte).leftMap(new DecoderError(s"unable to decode '$s' as a byte", _))
   implicit val shortDecoder: CellDecoder[Short] = s =>
-    Try(s.toShort).toEither.leftMap(new DecoderError(s"unable to decode '$s' as a short", _))
+    Either.catchNonFatal(s.toShort).leftMap(new DecoderError(s"unable to decode '$s' as a short", _))
   implicit val charDecoder: CellDecoder[Char] = s =>
-    if (s.size == 1)
+    if (s.length == 1)
       Right(s(0))
     else
       Left(new DecoderError(s"unable to decode '$s' as a character"))
   implicit val intDecoder: CellDecoder[Int] = s =>
-    Try(s.toInt).toEither.leftMap(new DecoderError(s"unable to decode '$s' as an integer", _))
+    Either.catchNonFatal(s.toInt).leftMap(new DecoderError(s"unable to decode '$s' as an integer", _))
   implicit val longDecoder: CellDecoder[Long] = s =>
-    Try(s.toLong).toEither.leftMap(new DecoderError(s"unable to decode '$s' as a long", _))
+    Either.catchNonFatal(s.toLong).leftMap(new DecoderError(s"unable to decode '$s' as a long", _))
   implicit val floatDecoder: CellDecoder[Float] = s =>
-    Try(s.toFloat).toEither.leftMap(new DecoderError(s"unable to decode '$s' as a float", _))
+    Either.catchNonFatal(s.toFloat).leftMap(new DecoderError(s"unable to decode '$s' as a float", _))
   implicit val doubleDecoder: CellDecoder[Double] = s =>
-    Try(s.toDouble).toEither.leftMap(new DecoderError(s"unable to decode '$s' as a double", _))
+    Either.catchNonFatal(s.toDouble).leftMap(new DecoderError(s"unable to decode '$s' as a double", _))
+  implicit val bigDecimalDecoder: CellDecoder[BigDecimal] = s =>
+    Try(BigDecimal(s)).toEither.leftMap(new DecoderError(s"unable to decode '$s' as a BigDecimal", _))
+  implicit val bigIntDecoder: CellDecoder[BigInt] = s =>
+    Try(BigInt(s)).toEither.leftMap(new DecoderError(s"unable to decode '$s' as a BigInt", _))
   implicit val stringDecoder: CellDecoder[String] = s => Right(s)
+  implicit val charArrayDecoder: CellDecoder[Array[Char]] = s => Right(s.toCharArray)
+
+  // Containers
+  implicit def decoderResultDecoder[T](implicit ev: CellDecoder[T]): CellDecoder[DecoderResult[T]] = s =>
+    ev(s).asRight
+  implicit def rawOrResultDecoder[T](implicit ev: CellDecoder[T]): CellDecoder[Either[String, T]] = s =>
+    ev(s).leftMap(_ => s).asRight
+  implicit def optionDecoder[T](implicit ev: CellDecoder[T]): CellDecoder[Option[T]] = s =>
```

Comment:
This is dangerous and might conflict with the `[Csv]RowDecoder` handling of optional columns, I think. Have you checked?